cabal-install/Distribution/Client/ProjectPlanOutput.hs

```diff
@@ -168,6 +169,59 @@ encodePlanAsJson distDirLayout elaboratedInstallPlan elaboratedSharedConfig =
             ] ++
             bin_file (compSolverName comp)
      where
+      packageLocationToJ :: PackageLocation (Maybe FilePath) -> J.Value
+      packageLocationToJ pkgloc =
+        case pkgloc of
+          LocalUnpackedPackage local ->
+            J.object [ "type" J..= J.String "local"
+                     , "path" J..= J.String local
+                     ]
+          LocalTarballPackage local ->
+            J.object [ "type" J..= J.String "local-tarball"
+                     , "path" J..= J.String local
+                     ]
+          RemoteTarballPackage uri _ ->
+            J.object [ "type" J..= J.String "remote-tarball"
+                     , "uri"  J..= J.String (show uri)
+                     ]
+          RepoTarballPackage repo _ _ ->
+            J.object [ "type" J..= J.String "repo-tarball"
+                     , "repo" J..= repoToJ repo
+                     ]
+          RemoteSourceRepoPackage srcRepo _ ->
+            J.object [ "type" J..= J.String "source-repo"
+                     , "source-repo" J..= sourceRepoToJ srcRepo
+                     ]
+
+      repoToJ :: Repo -> J.Value
+      repoToJ repo =
+        case repo of
+          RepoLocal{..} ->
+            J.object [ "type" J..= J.String "local-repo"
+                     , "path" J..= J.String repoLocalDir
+                     ]
+          RepoRemote{..} ->
+            J.object [ "type"   J..= J.String "remote-repo"
+                     , "path"   J..= J.String repoLocalDir
+                     , "remote" J..= J.String (remoteRepoName repoRemote)
+                     ]
+          RepoSecure{..} ->
+            J.object [ "type"   J..= J.String "secure-repo"
+                     , "path"   J..= J.String repoLocalDir
```

Comment:
do we really need the `"path"`? I don't feel like that's a relevant information; we already know the "remote" label; also the local repo cache path is to be considered an implementation detail; its on-disk format may easily change at some point to account for situations we can't currently represent properly yet.

Reply:
yeah, let's remove that one.